python/tank/bootstrap/manager.py

```diff
@@ -626,35 +626,68 @@ def _is_toolkit_activated_in_shotgun(self):
         log.debug("...enabled: %s" % enabled)
         return enabled
 
-    def _cache_apps(self, tk, progress_callback, do_post_install=False):
+    def _cache_apps(self, tk, config_engine_name, config_project_id, progress_callback, do_post_install=False):
         """
         Caches all apps associated with the given toolkit instance.
 
         :param tk: Bootstrapped :class:`~sgtk.Sgtk` instance to cache items for.
+        :param config_engine_name: Name of the engine that was used to resolve the configuration.
+        :param config_project_id: Project id that was used to resolve the configuration;
+                                  None for the site configuration.
         :param progress_callback: Callback function that reports back on the engine startup progress.
         :param do_post_install: Set to true to execute the post install triggers.
         """
 
+        from ..platform import constants as platform_constants
+
         log.info("Downloading and installing apps...")
 
+        # Resolve a context for the project.
+        if config_project_id:
+            project_entity = {"type": "Project", "id": config_project_id}
+            project_context = tk.context_from_entity_dictionary(project_entity)
+        else:
+            project_context = tk.context_empty()
```

Comment:
minor: maybe it's clearer to just call this context and not project_context? It doesn't always hold a project.